### DISCUSSION TITLE
Do i still need separate backend?

### DISCUSSION THREAD

**Discussion by a0s** [reflex-dev/reflex#2961] [General]
My question may seem a little strange. But.. After reading the documentation, I still don’t fully understand whether I can refuse a separate backend or not. Typical tasks that we solve on the backend: authorization (jwt for e,g.), working with the database, validating data from frontend, creating and processing jobs in the mq queue, etc. 
Does it make sense to try to fit all these things inside reflex handlers? It is clear that all slow things must be moved to separate asyncs/threads..
~It’s also not entirely clear whether I can use the same instance (let’s call it reflex-backend) to host the main backend and how the code should be organized in this case?~ Seems i missed https://reflex.dev/docs/api-routes/overview/#reserved-routes

**Comment by picklelo** [comment]
@a0s check out our [blog](https://reflex.dev/blog/2024-03-21-reflex-architecture/)on how Reflex works under the hood to get a better idea.

The short answer is no, you don't need a separate backend. You can think of the state/event handlers as the backend of your app, since these run on the server. All your database connections and jobs etc. can just be called from these event handlers. We have [background events](https://reflex.dev/docs/events/background-events) for long running tasks that don't block the frontend and other events.
